reader_test.go

```diff
@@ -1047,6 +1053,54 @@ func testReaderConsumerGroupReadContentAcrossPartitions(t *testing.T, ctx contex
 	}
 }
 
+// Build a struct to implement the ReadPartitions interface.
+type MockConnWatcher struct {
+	count int
+	partitions [][]Partition
+}
+func (m *MockConnWatcher) ReadPartitions(topics ...string) (partitions []Partition, err error) {
+	partitions = m.partitions[m.count]
+	m.count++
+	return partitions, err
+}
+
+func testReaderConsumerGroupRebalanceOnPartitionAdd(t *testing.T, ctx context.Context, r *Reader) {
+	// Sadly this test is time based, so at the end will be seeing if the runGroup run to completion within the
+	// allotted time. The allotted time is 2x the PartitionWatchInterval.
+	now := time.Now()
+	watchTime := 500 * time.Millisecond
+	conn := &MockConnWatcher{
+		partitions: [][]Partition{
+			{
+				Partition{
+					Topic: "topic-1",
+					ID:    0,
+				},
+			},
+			{
+				Partition{
+					Topic: "topic-1",
+					ID:    0,
+				},
+				{
+					Topic: "topic-1",
+					ID:    1,
+				},
+			},
+		},
+	}
+
+	rg := &runGroup{}
+	rg = rg.WithContext(ctx)
+	r.config.PartitionWatchInterval = watchTime
+	rg.Go(r.partitionWatcher(conn))
+	rg.Wait()
+	if time.Now().Sub(now).Seconds() > r.config.PartitionWatchInterval.Seconds() * 2 {
```

Comment:
If you make the change above, you can add in additional padding here.  Since there's a lot of parallel tests going on, it's usually best to add additional time.